apps/studio/components/interfaces/Organization/GeneralSettings/OrganizationDetailsForm.tsx

```diff
@@ -0,0 +1,136 @@
+
+// Schema for organization details using Zod
+const OrgDetailsSchema = z.object({
+  name: z.string().min(1, 'Organization name is required'),
+})
+
+const OrganizationDetailsForm = () => {
+  const selectedOrganization = useSelectedOrganization()
+  const canUpdateOrganization = useCheckPermissions(PermissionAction.UPDATE, 'organizations')
+  const { slug } = useParams()
+  const queryClient = useQueryClient()
+
+  const { mutate: updateOrganization, isLoading: isUpdatingDetails } =
+    useOrganizationUpdateMutation()
+
+  const orgDetailsForm = useForm<z.infer<typeof OrgDetailsSchema>>({
+    resolver: zodResolver(OrgDetailsSchema),
+    defaultValues: { name: selectedOrganization?.name ?? '' },
+  })
+
+  useEffect(() => {
+    if (selectedOrganization && !isUpdatingDetails) {
+      orgDetailsForm.reset({ name: selectedOrganization.name ?? '' })
+    }
+  }, [selectedOrganization, orgDetailsForm.reset, isUpdatingDetails])
+
+  const onUpdateOrganizationDetails = async (values: z.infer<typeof OrgDetailsSchema>) => {
+    if (!canUpdateOrganization) {
+      return toast.error('You do not have the required permissions to update this organization')
+    }
+    if (!slug) return console.error('Slug is required')
+
+    updateOrganization(
+      { slug, name: values.name },
+      {
+        onSuccess: () => {
+          invalidateOrganizationsQuery(queryClient)
+          toast.success('Successfully updated organization name')
+        },
+        onError: (error: ResponseError) => {
+          toast.error(`Failed to update organization name: ${error.message}`)
+        },
+      }
+    )
+  }
+
+  const permissionsHelperText = !canUpdateOrganization
+    ? "You need additional permissions to manage this organization's settings"
+    : undefined
+
+  return (
+    <Form_Shadcn_ {...orgDetailsForm}>
+      <form
+        id="org-details-form"
+        onSubmit={orgDetailsForm.handleSubmit(onUpdateOrganizationDetails)}
+      >
+        <Card>
+          <CardContent className="pt-6">
+            <FormField_Shadcn_
+              control={orgDetailsForm.control}
+              name="name"
+              render={({ field }) => (
+                <FormItemLayout label="Organization name" layout="flex-row-reverse">
+                  <FormControl_Shadcn_>
+                    <Input
+                      {...field}
+                      className="w-96 max-w-full"
+                      disabled={!canUpdateOrganization || isUpdatingDetails}
+                    />
+                  </FormControl_Shadcn_>
+                </FormItemLayout>
+              )}
+            />
+          </CardContent>
+          <CardContent>
+            <FormItemLayout label="Organization slug" layout="flex-row-reverse">
+              <PrePostTab
+                postTab={
+                  <CopyButton type="text" iconOnly text={selectedOrganization?.slug ?? ''} />
+                }
+              >
+                <Input
```

Comment:
nit feedback: width of the input doesn't need to be this long, given that the org slug is a fixed length text
![image](https://github.com/user-attachments/assets/1f7d8d60-626c-459b-9350-59cfb03db6cc)